spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -99,7 +99,8 @@ class SampleToBatch[T: ClassTag]
 (totalBatch : Int,
  featurePadding : Option[Tensor[T]] = None,
  labelPadding : Option[T] = None,
- fixedLength: Option[Int] = None)
+ fixedLength: Option[Int] = None,
+ batchCore : Option[Boolean] = None)
```

Comment:
SampleToBatch has to support both train and predict. Total batch means batch per core in predict, but means batch per node in train stage. So add a flag to distinguish.